lib/Sema/CSRanking.cpp

```diff
@@ -1405,6 +1405,10 @@ SolutionCompareResult ConstraintSystem::compareSolutions(
 Optional<unsigned>
 ConstraintSystem::findBestSolution(SmallVectorImpl<Solution> &viable,
                                    bool minimize) {
+  // Don't spend time filtering solutions if we already hit a threshold.
+  if (isTooComplex(viable))
```

Comment:
One last thing, I promise! :) Please add `isTooComplex(/*solutions=*/{})` check to each `for` loop in the body of `findBestSolution` that way we can make sure that solver times out while working through comparisons because that is n^3 algorithm.

Reply:
IMO `ArrayRef` is unnecessary indirection since there is a definite container type, but if you insist.

Reply:
I hear what you are saying but it doesn't really cost much and it's safer API.